planetiler-core/src/main/java/com/onthegomap/planetiler/files/WriteableFilesArchive.java

```diff
@@ -0,0 +1,92 @@
+package com.onthegomap.planetiler.files;
+
+import com.google.common.base.Preconditions;
+import com.onthegomap.planetiler.archive.TileEncodingResult;
+import com.onthegomap.planetiler.archive.WriteableTileArchive;
+import com.onthegomap.planetiler.geo.TileOrder;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class WriteableFilesArchive implements WriteableTileArchive {
+
+  private final Path basePath;
+
+  private WriteableFilesArchive(Path basePath) {
+    this.basePath = basePath;
+    if (!Files.exists(basePath)) {
+      mkdirs(basePath);
+    }
+    Preconditions.checkArgument(
+      Files.isDirectory(basePath),
+      "require \"" + basePath + "\" to be a directory"
+    );
+  }
+
+  public static WriteableFilesArchive newWriter(Path basePath) {
+    return new WriteableFilesArchive(basePath);
+  }
+
+  @Override
+  public boolean deduplicates() {
+    return false;
+  }
+
+  @Override
+  public TileOrder tileOrder() {
+    return TileOrder.TMS;
+  }
+
+  @Override
+  public TileWriter newTileWriter() {
+    return new FilesWriter(basePath);
+  }
+
+  @Override
+  public void close() throws IOException {
+    // nothing to do here
+  }
+
+  private static void mkdirs(Path p) {
+    try {
+      Files.createDirectories(p);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private static class FilesWriter implements TileWriter {
+
+    private final Path basePath;
+    private Path lastCheckedFolder;
+
+    FilesWriter(Path basePath) {
+      this.basePath = basePath;
+      this.lastCheckedFolder = basePath;
+    }
+
+    @Override
+    public void write(TileEncodingResult encodingResult) {
+
+      final Path file = FilesArchiveUtils.absolutePathFromTileCoord(basePath, encodingResult.coord());
+      final Path folder = file.getParent();
+
+      // tiny optimization in order to avoid too many unnecessary "folder-exists-checks" (I/O)
+      if (!lastCheckedFolder.equals(folder) && !Files.exists(folder)) {
+        mkdirs(folder);
+      }
+      lastCheckedFolder = folder;
+      try {
+        Files.write(file, encodingResult.tileData());
```

Comment:
`encodingResult.tileData()` is gzipped here. We should probably make it configurable whether or not the output files should be gzipped (I think default to false?). If they aren't gzipped we can skip the gzip step while encoding, and if they are gzipped the output extension should probably be `.pbf.gz`

Reply:
The output is not necessarily gzipped. This is already configurable at a higher level: tile_compression. So I don't think anything needs to be handled here specifically at the archive level.

Reply:
I personally prefer the tiles to be at a known location by default. With the introduction of the tile_scheme param users can adjust as needed.
https://github.com/onthegomap/planetiler/commit/2e4652ef39e501b472265852c261d7685cf179ee